dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/serializer/GsonCompatibilityTests.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.serializer
+
+import com.datadog.tools.unit.extensions.TestConfigurationExtension
+import com.google.gson.JsonObject
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.RepeatedTest
+import org.junit.jupiter.api.Test
+
+/**
+ * This class is used to test the compatibility of the Gson serializer with our current `.`
+ * keys in the JSON payloads. This should make sure that Json properties are added into the
+ * `toString` representation in the order they were added in the JSON object.
+ */
+
+@Extensions(ExtendWith(ForgeExtension::class), )
+internal class GsonCompatibilityTests {
+
+    @RepeatedTest(5)
+    fun `M add the properties in order in the string W toString()`(forge:Forge){
+        val propertiesKeysValues = forge.aMap(size = forge.anInt(min=5,max=50)){
+                val key = forge.aList(size = forge.anInt(min = 1, max = 10)) {
+                    anAlphabeticalString() }.joinToString(".")
```

Comment:
Do we need to have dots in keys? We simply test that keys are added in order, we don't care about keys format.

Reply:
yes but I wanted to just make sure it is robust enough. It doesn't hurt to have the dots there.